#2664 Added xid validation for event detectors

Patrykb0802 opened the pull request:
- Added maxlength="50" parameter to Export ID (XID) field

Patrykb0802 committed:
#2664 Added xid validation for event detectors

- Added maxlength="50" parameter to Export ID (XID) field

Patrykb0802 committed:
#2664 Added xid validation for event detectors

- Added maxlength="50" parameter to Export ID (XID) for every event detector

- Added 'if' in validate function inside DataPointEditController.java that checks if XID is longer than 50 characters, if it is than saving breaks and notification pops up

- Added new key for translations (validate.ped.xidTooLong=XID cannot be longer than 50 characters)

Patrykb0802 committed:
#2664 Added xid validation for event detectors

- Added two public static validatesXid methods in XidUtils.java:

validateXid(Map<String, String> errors, ResourceBundle resourceBundle, PointEventDetectorVO ped) and

validateXid(DwrResponseI18n response, String xid)

- Replace code that was using similar validation to methods mentioned above with validateXid method

Patrykb0802 committed:
- Deleted unnecessary commented method

Patrykb0802 committed:
- Formatting adjustments

Limraj commented on webapp-resources/messages_fi.properties:
```diff
@@ -2339,6 +2339,7 @@ validate.chart.incompatible=Kaavion muotoilu ei ole yhteensopiva datatyypille
 validate.changeType.incompatible=Muutostyyppi ei ole yhteensopiva datatyypille
 validate.axis.invalid=Pit\u00e4\u00e4 olla jokin n\u00e4ist\u00e4 A, B, C, D, E, F, G, tai H
 validate.ped.xidUsed=Sama XID ''{0}'' on k\u00e4yt\u00f6ss\u00e4 useille tapahtumille. Ei tallennettu
+validate.ped.xidTooLong=XID ei voi olla pidempi kuin 50 merkki�.
```
We leave in English.

Limraj commented on webapp-resources/messages_de.properties:
```diff
@@ -2978,6 +2978,7 @@ validate.parameterRequired=Parameter name is required
 validate.parseError=Lesefehler: {0}
 validate.ped.xidMissing=All event detectors must define an XID. Not saved
 validate.ped.xidUsed=The same XID ''{0}'' is being used for multiple event detectors. Not saved
+validate.ped.xidTooLong=XID cannot be longer than 50 characters
```
The validate.ped.xidTooLong key was supposed to be deleted, the key exists: validate.notLongerThan and  validate.required.

Limraj reviewed (commented):
Proposal for an amendment. Full xid validation in one place.
[XidUtils_java.patch](https://github.com/SCADA-LTS/Scada-LTS/files/12690628/XidUtils_java.patch)

Patrykb0802 committed:
#2664 Added xid validation for event detectors

- Moved xidUnique validation to XidUtils.java

- In DataSourceVO.java added error handling, in XidUtils.java validation is using xid contextKey, in DataSourceVO.java used contextKey is dataSourceXid, so displaying errors was added.

Tested:

- XID handling in publisher, watchlist, datasource edit and datapoint edit views - OK

Patrykb0802 committed:
#2664 Added xid validation for event detectors

- Added validateXid method for different needs of form validation
